py-polars/polars/expr/expr.py

```diff
@@ -3303,64 +3305,78 @@ def cut(
         Parameters
         ----------
         breaks
-            A list of unique cut points.
+            List of unique cut points.
         labels
-            Labels to assign to bins. If given, the length must be len(breaks) + 1.
+            Names of the categories. The number of labels must be equal to the number
+            of cut points plus one.
         left_closed
-            Whether intervals should be [) instead of the default of (]
+            Set the intervals to be left-closed instead of right-closed.
         include_breaks
-            Include the the right endpoint of the bin each observation falls in.
-            If True, the resulting column will be a Struct.
+            Include a column with the right endpoint of the bin each observation falls
+            in. This will change the data type of the output from a
+            :class:`Categorical` to a :class:`Struct`.
+
+        Returns
+        -------
+        Expr
+            Expression of data type :class:`Categorical` if ``include_breaks`` is set to
+            ``False`` (default), otherwise an expression of data type :class:`Struct`.
+
+        See Also
+        --------
+        qcut
 
         Examples
         --------
-        >>> g = pl.repeat("a", 5, eager=True).append(pl.repeat("b", 5, eager=True))
-        >>> df = pl.DataFrame(dict(g=g, x=range(10)))
-        >>> df.with_columns(q=pl.col("x").cut([2, 5]))
-        shape: (10, 3)
-        ┌─────┬─────┬───────────┐
-        │ g   ┆ x   ┆ q         │
-        │ --- ┆ --- ┆ ---       │
-        │ str ┆ i64 ┆ cat       │
-        ╞═════╪═════╪═══════════╡
-        │ a   ┆ 0   ┆ (-inf, 2] │
-        │ a   ┆ 1   ┆ (-inf, 2] │
-        │ a   ┆ 2   ┆ (-inf, 2] │
-        │ a   ┆ 3   ┆ (2, 5]    │
-        │ …   ┆ …   ┆ …         │
-        │ b   ┆ 6   ┆ (5, inf]  │
-        │ b   ┆ 7   ┆ (5, inf]  │
-        │ b   ┆ 8   ┆ (5, inf]  │
-        │ b   ┆ 9   ┆ (5, inf]  │
-        └─────┴─────┴───────────┘
```

Comment:
I'd suggest keeping some of the old examples like this around. It's helpful to understand the binning behavior, especially with different endpoint rules.

Reply:
I thought the old examples were unclear. The example should be about `cut`, not showcasing other Polars functionality.

In this example, people unfamiliar with `repeat`/`append` might not know what the original data looks like. And then after the cut operation, we cannot even see the whole dataframe.

The data must be short and simple, the only complexity should be in the method we're trying to demo. See the README in the `py-polars/tests` folder for more context.

Reply:
Isn't the point that it's pretty easy to run it? I agree about not seeing the whole frame but it's hard to make a demonstration of something that discretizes data, especially when it's potentially across groups and you only have a few rows. Probably ok to remove the "over" behavior since I only had that because it was an issue at the time. But for better or worse it's somewhat complex and the binning behavior needs to be shown since just having names doesn't make the left/right closed distinction clear and making the leftmost and rightmost endpoints `-inf` and `inf` could actually be surprising to some (this might need to change).

Reply:
Please see my message on Discord!